src/types.ts

```diff
@@ -1,14 +1,22 @@
 import type { PlatformProxy } from "wrangler";
 
+type IncomingRequestCfProperties<T> =
+  T extends PlatformProxy<any, infer T> ? T : never;
+
+export interface CfProperties
+  extends IncomingRequestCfProperties<PlatformProxy<any, any>> {}
+
+export interface Env {}
+
+type TypedPlatformProxy = PlatformProxy<Env, CfProperties>;
```

Comment:
/cc @dario-piotrowicz Do you think we can upstream this for wrangler types?

Reply:
Yeah we probably should use `unknown`. I let @
Aslemammad  to explain more his usage context.

Reply:
I just added CfProperties since it was a generic so maybe a user would want this. But I guess it's not needed as I understand.

Let me remove it.

Reply:
@pi0 sorry it took me a while to come back to you on this one 🙇 

By upstreaming the change, is this along the lines of what you had in mind?
https://github.com/dario-piotrowicz/nitro-generic-wrangler-env-type-example

Reply:
> I just added CfProperties since it was a generic so maybe a user would want this. But I guess it's not needed as I understand.
> 
> Let me remove it.

@Aslemammad yeah sorry that made sense, I think that the type that `getPlatformProxy` uses by default is anyways good and people should really not have a need to tweak that often 🤔 

It might still be valuable to have it generic... if it were up to me I'd suggest keeping it as is and make it generic/user configurable if the need arises, WDYT? 🤔

Reply:
I believe with the upstream change, we might have to close this PR!

Reply:
Thank you so much for the more robust implementation.

Reply:
No problem @Aslemammad, it's my pleasure 🙂 

> I believe with the upstream change, we might have to close this PR!

Yes ideally with the upstream change we should be able to close this PR, but before doing that lets see if the wrangler team accepts my PR as I imagine that the `Env` type there can be slightly controversial and get some pushback (as it could arguably live in `@cloudflare/workers-types` instead)

Reply:
I appreciate you @dario-piotrowicz! Ok then, let's wait and see what happens.